src/gui.ml

```diff
@@ -154,11 +155,51 @@ let draw_position (bd : Game.t) (imgs : image list) =
   let bd = board_to_array bd in
   draw_position_rows bd imgs 0
 
-let draw_game (bd : Game.t) =
+let rec get_potential_squares (move_list : move list) currX currY :
+    (int * int) list =
+  match move_list with
+  | ((a, b), (c, d)) :: t when currX = a && currY = b ->
+      (c, d) :: get_potential_squares t currX currY
+  | h :: t -> get_potential_squares t currX currY
+  | _ -> []
+
+let draw_potential_circles
```

Comment:
Do you think we should just combine draw_circles and draw_potential_circles into one function? They share the same inputs and it wouldn't be very long so it seems right.